test/modules/CloudToDeviceMessageTester/Receiver.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace CloudToDeviceMessageTester
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Extensions.Logging;
+    using TransportType = Microsoft.Azure.Devices.TransportType;
+
+    sealed class Receiver : CloudToDeviceMessageTesterBase
+    {
+        DeviceClient deviceClient;
+
+        public Receiver(
+            ILogger logger,
+            string iotHubConnectionString,
+            string deviceId,
+            string moduleId,
+            TransportType transportType,
+            TimeSpan testDuration,
+            TestResultReportingClient testResultReportingClient)
+            : base(logger, iotHubConnectionString, deviceId, moduleId, transportType, testDuration, testResultReportingClient)
+        {
+        }
+
+        public override void Dispose() => this.deviceClient?.Dispose();
+
+        public async Task ReportTestResult(Message message)
+        {
+            string trackingId = message.Properties[TestConstants.Message.TrackingIdPropertyName];
+            string batchId = message.Properties[TestConstants.Message.BatchIdPropertyName];
+            string sequenceNumber = message.Properties[TestConstants.Message.SequenceNumberPropertyName];
+            var testResultReceived = new MessageTestResult(this.moduleId + ".receive", DateTime.UtcNow)
+            {
+                TrackingId = trackingId,
+                BatchId = batchId,
+                SequenceNumber = sequenceNumber
+            };
+            await ModuleUtil.ReportTestResultAsync(this.testResultReportingClient, this.logger, testResultReceived);
+        }
+
+        public override async Task InitAsync(CancellationTokenSource cts, DateTime testStartAt)
+        {
+            Microsoft.Azure.Devices.RegistryManager registryManager = null;
+            try
+            {
+                registryManager = Microsoft.Azure.Devices.RegistryManager.CreateFromConnectionString(this.iotHubConnectionString);
+                await registryManager.AddDeviceAsync(new Microsoft.Azure.Devices.Device(this.deviceId + "_leaf"), cts.Token);
```

Comment:
Isn't there an upper limit of how long a deviceName can be? Please double check me and make sure that by adding "_leaf" would not exceed the current deviceName length.

Reply:
Apparently it should be fine with this length, as it is quite short. If there is a problem we will catch it at the very start of the test.